src/pages/index.js

```diff
@@ -7,96 +7,208 @@ import Card from "../components/Card.js";
 
-const newCardPopup = new PopupWithForm(cardModalSelector, submitCard);
-const previewImagePopup = new PopupWithImage(previewImageModal);
+profileEditButton.addEventListener("click", openProfilePopup);
 
-addNewCardButton.addEventListener("click", () => {
-  addCardFormValidator.toggleButtonState();
-  newCardPopup.open();
+profileAvatarPeniclIcon.addEventListener("click", () => {
+  avatarFromValidator.toggleButtonState();
+  editAvatarPopup.open();
+  editAvatarPopup.setLoading(false, "Save");
 });
 
 /* ---------------------------------------------------------------------------------------------- */
 /*                                         Card Functions                                         */
 /* ---------------------------------------------------------------------------------------------- */
 
-function createCard({ name, link }) {
-  const cardElement = new Card({ name, link }, "#card-template", ({ name, link }) => {
-    previewImagePopup.open({ name, link });
-  });
-  return cardElement.generateCard();
+//! Do Not Delete
+function handleProfileAvatarSubmit(url) {
+  api
+    .setUserAvatar(url)
+    .then((userData) => {
+      editAvatarPopup.setLoading(true);
+      userInfo.setProfileAvatar(userData.avatar);
+      editAvatarPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
 }
 
-function submitCard({ title, url }) {
-  const newCardData = { name: title, link: url };
-  const newCard = createCard(newCardData); // creates "newCard" & stores the function returned execution of createCard fn
-  cardListSection.prependItem(newCard); // prepend method from Section class
-  newCardPopup.close();
+//! Do not delete
+function handleSubmitCard({ title, url }) {
+  api
+    .addCard(title, url)
+    .then((card) => {
+      const newCard = createCard(card);
+      cardListElement.prepend(newCard);
+      addNewCardPopup.setLoading(true);
+      addNewCardPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
 }
+
+//! Do not delete
+function createCard(data) {
+  const newCard = new Card(
+    data,
+    userId,
+    cardTemplateElement,
+    function handleCardClick() {
+      previewImagePopup.open(data);
+    },
+    function handleCardDelete() {
+      deleteImagePopup.setLoading(false, "Yes");
+      deleteImagePopup.setSubmitAction(() => {
+        api
+          .deleteCard(data._id)
+          .then((res) => {
+            newCard.remove(res._id);
+            deleteImagePopup.setLoading(true);
+            deleteImagePopup.close();
+          })
+          .catch((err) => {
+            console.error(err);
+          });
+      });
+      deleteImagePopup.open(data._id);
+    },
+    function handleCardLikeClick(data) {
+      api
+        .changeLikeCardStatus(data._id, newCard.isLiked())
+        .then((res) => {
+          console.log(res);
+          const likes = res.likes || [];
+          newCard.setLikes(likes);
+          if (this.isLiked()) {
```

Comment:
`this` should **not** be used outside classes in functions in `index.js` . This context is easy to lose. You need to find other ways of passing data.



```js

newCard.isLiked()
```